File: TwoMotorsArdu/main/Command1.cpp
#include "Command1.hpp"

Command1::Command1() : pwm(0), direction(Direction::forward) {}

Command1::Command1(int pwm, Direction direction) : pwm(pwm), direction(direction) {}

int Command1::getPwm() {
    return pwm;
}

Direction Command1::getDirection() {
    return direction;
}

File: ts-TwoMotors/Command1.ts
export class Command1{
    private pwm : number;
    private direction : Direction;
    constructor(obj:any){
        this.pwm = obj.pwm;
        this.direction = obj.direction;
    }
    public getData():Uint8Array{
        const pwmByte : number = this.pwm & 0xFF;
        const directionByte : number = Number(this.direction) & 0xFF;
        return new Uint8Array([pwmByte, directionByte]);
    }
}